File: typescript-client-sdk/src/sdk/models/operations/getonecrmeventattendee.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmEventAttendeeQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmEventAttendeeResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=eventAttendee" })
  eventAttendee?: shared.EventAttendee;
}


export class GetOneCrmEventAttendeeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmEventAttendeeQueryParams;
}


export class GetOneCrmEventAttendeeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmEventAttendeeResponseBody;
}

File: go-client-sdk/pkg/models/operations/getonecrmeventattendee.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmEventAttendeeQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmEventAttendeeResponseBody struct {
	EventAttendee *shared.EventAttendee `json:"eventAttendee,omitempty"`
}

type GetOneCrmEventAttendeeRequest struct {
	QueryParams GetOneCrmEventAttendeeQueryParams
}

type GetOneCrmEventAttendeeResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmEventAttendeeResponseBody
}
